src/state.py

```diff
@@ -5,11 +5,54 @@ def __init__(self):
     def add_student(self, student):
         self.students.append(student)
 
+    #Given two students and a subject, trades their classes for that given subject
+    def trade_classes(self,student1,student2,subject_name):
+        student1_class = student1.get_class_for_subject(subject_name)
+        student2_class = student2.get_class_for_subject(subject_name)
+        student1.set_class_for_subject(subject_name, student2_class)
+        student2.set_class_for_subject(subject_name, student1_class)
+        #student1.remove_subject_target(subject_name) #E suposto ele remover os targets a medida que vai trocando? Ou e usado para a funçao de fitness?
```

Comment:
Replying to your commented question:
No, we should not delete the targets nor the give ins as a student moves to that class. 

Let's say that student A
- Has a class B in subject Z.
- Has as target class C and D for subject Z.  
  
Student A trades class B for class C and C is eliminated from the targets.
Student A trades class C for class D and D is eliminated from the targets.
Now, let's say that some time after student A gets class D in subject Z, there exists a neighbor with better fitness which needs student A to get class C in subject Z. 

If we delete the subject targets, we can no longer achieve a state were student A has class C in subject Z.